### ISSUE TITLE
Does Pinot support string split function?

### ISSUE THREAD

**Issue by Lvnszn** [apache/pinot#12089]
Pinot version: 1.0.0 / 0.12.1

SQL: select split('test, tt', ',', 2) as t from customer_flow_stat

Result:
![image](https://github.com/apache/pinot/assets/12541438/ead7fc6e-0c32-4835-a4b5-2bbc810f68ba)

**Comment by walterddr** [comment]
yes it is supported. but seems like the support for literal evaluation is missing. 
upon checking the code I think it is because  https://github.com/apache/pinot/blob/b25f7cf6c165d9de39d79e86894235fca359adab/pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java#L220 doesn't understand array type and thus default stringify using java's toString method and thus output the object ref string

**Comment by xiangfu0** [comment]
right, we should either push this to server for processing or support array literal as result holder

**Comment by xiangfu0** [comment]
https://github.com/apache/pinot/pull/12118
